src/Policies.cpp

```diff
@@ -174,6 +174,18 @@ std::vector<Policy *> PolicyManager::load(const YAML::Node &policy_yaml)
             window_config.config_set<uint64_t>("deep_sample_rate", _default_deep_sample_rate);
         }
 
+        std::unique_ptr<Policy> input_resources_policy;
+        std::unique_ptr<StreamHandler> resources_module;
+        if (input_stream) {
```

Comment:
since the input_stream may be being reused here, it needs to check if the input_resources_policy has already been added to this input_stream, right? otherwise we'd get > 1

Reply:
This input_stream is the unique pointer. it only exists when new input is create, otherwise we will only have the input_ptr.